src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -144,6 +144,10 @@ def add(self, new_symbol, tag=None):
         :raises KeyError: if the symbol name is already in use.
 
         '''
+        if not isinstance(new_symbol, Symbol):
+            raise InternalError("Symbol '{0}' is not a symbol, but '{1}'.'"
+                                .format(new_symbol, type(new_symbol)))
```

Comment:
If you do `type(new_symbol).__name__` then the result is independent of whether you're using Python 2 or 3.

Reply:
I didn't know that, that's great.